tempest-testing-junit4/src/main/kotlin/app/cash/tempest/testing/TestDynamoDb.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.tempest.testing
+
+import org.junit.rules.ExternalResource
+import java.util.concurrent.ConcurrentHashMap
+
+class TestDynamoDb private constructor(
```

Comment:
Document that it stays running?

Reply:
I wonder if you could get away with a single JUnit4 and JUnit5 extension by using only Guava service as the public API?

There’s nothing particularly Dynamo2 or Dynamo1 specific in this, just the types it imports.

Reply:
We do need 2 different `TestTable` classes tho to allow developers to create GSIs in SDK 1 and 2